client/src/modules/multiple_payroll/modals/search.modal.js

```diff
@@ -0,0 +1,126 @@
+angular.module('bhima.controllers')
+  .controller('MultiPayrollSearchModalController', MultiPayrollSearchModalController);
+
+MultiPayrollSearchModalController.$inject = [
+  '$uibModalInstance', 'filters', 'NotifyService', 'Store', 'util',
+  'MultiplePayrollService', 'CurrencyService', 'PayrollConfigurationService', '$translate',
+];
+
+/**
+ * @class MultiPayrollSearchModalController
+ *
+ * @description
+ * This controller is responsible to collecting data from the filter form and
+ * returning it as a JSON object to the parent controller.  The data can be
+ * preset by passing in a filters object using filtersProvider().
+ */
+function MultiPayrollSearchModalController(
+  ModalInstance, filters, Notify, Store, util,
+  MultiplePayroll, Currencies, Payroll, $translate
+) {
+  const vm = this;
+  const changes = new Store({ identifier : 'key' });
+  const searchQueryOptions = [
+    'payroll_configuration_id', 'currency_id', 'display_name', 'code', 'status_id'
+  ];
+  const lastValues = {};
+
+  let statusText = '/';
+
+  // displayValues will be an id:displayValue pair
+  const displayValues = {};
+  const lastDisplayValues = MultiplePayroll.filters.formatView().defaultFilters;
+
+
+  lastDisplayValues.forEach((last) => {
```

Comment:
Just FYI - I think this is already implemented in [FilterService.prototype.getDisplayValueMap()](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/client/src/js/services/FilterService.js#L200).